demo/src/plot_demo.rs

```diff
@@ -1101,6 +1107,61 @@ impl ChartsDemo {
     }
 }
 
+#[derive(PartialEq, serde::Deserialize, serde::Serialize, Default)]
+struct UserdataDemo {}
+
+struct DemoPoint {
+    x: f64,
+    y: f64,
+    custom_thing: bool,
+}
+
+impl UserdataDemo {
+    #[allow(clippy::unused_self, clippy::significant_drop_tightening)]
+    fn ui(&self, ui: &mut egui::Ui) -> Response {
+        let points = (1..=1000)
+            .map(|i| DemoPoint {
+                x: i as f64 / 1000.0,
+                y: ((i as f64) / 1000.0 * std::f64::consts::PI * 2.0).sin(),
+                custom_thing: i % 2 == 0,
+            })
+            .collect::<Vec<_>>();
+
+        let custom_things =
+            std::sync::Arc::new(egui::mutex::Mutex::new(std::collections::HashMap::<
```

Comment:
Add some `use` statements to the top of this file so this becomes easier to read